test/unit/spec/PollController_spec.js

```diff
@@ -72,47 +72,49 @@ function (Okta, Q, OktaAuth, Util, PollingForm, Expect, Router, $sandbox, resPol
             Expect.isJsonPost($.ajax.calls.argsFor(0), {
               url: 'https://example.okta.com/api/v1/authn/cancel',
               data: {
-                stateToken: '00_J1qxqyLs-6ZutUUWfbqm-1nqnW6n2o5z2wnBRHs'
+                stateToken: '00caLtr2zn6rEXWaHyNNmTfczuToPNY2R8y3f0yAK_'
               }
             });
           });
       });
     });
   });
 
-  Expect.describe('Polling', function () {
-    describe('API', function () {
-      itp('starts polling on load', function () {
-        return setup({}, [resPolling, resPolling, resPolling, resSuccess]).then(function (test) {
-          return Expect.wait(function () {
-            return $.ajax.calls.count() > 3;
-          }, test);
-        })
-          .then(function () {
+  Expect.describe('Polling API', function () {
+    describe('called on load', function () {
+      var originalTimeout;
+      beforeEach(function () {
+        originalTimeout = jasmine.DEFAULT_TIMEOUT_INTERVAL;
+        jasmine.DEFAULT_TIMEOUT_INTERVAL = 50000;
+      });
+
+      it('makes the request correctly', function (done) {
+        setup({}, [resPolling, resPolling, resSuccess]).then(() => {
+          setTimeout(function () {
+            expect($.ajax).toHaveBeenCalledTimes(3);
             // first call is for refresh-auth
-            expect($.ajax).toHaveBeenCalledTimes(4);
-            // 1st poll
+            // poll
             Expect.isJsonPost($.ajax.calls.argsFor(1), {
-              url: 'https://example.okta.com/api/v1/authn/factors/okta-poll/poll',
+              url: 'https://example.okta.com/api/v1/authn/poll',
               data: {
-                stateToken: '00_J1qxqyLs-6ZutUUWfbqm-1nqnW6n2o5z2wnBRHs'
+                stateToken: '00caLtr2zn6rEXWaHyNNmTfczuToPNY2R8y3f0yAK_'
               }
             });
             // 2nd poll
             Expect.isJsonPost($.ajax.calls.argsFor(2), {
-              url: 'https://example.okta.com/api/v1/authn/factors/okta-poll/poll',
+              url: 'https://example.okta.com/api/v1/authn/poll',
               data: {
-                stateToken: '00_J1qxqyLs-6ZutUUWfbqm-1nqnW6n2o5z2wnBRHs'
+                stateToken: '00caLtr2zn6rEXWaHyNNmTfczuToPNY2R8y3f0yAK_'
               }
             });
-            // 3rd poll
-            Expect.isJsonPost($.ajax.calls.argsFor(3), {
-              url: 'https://example.okta.com/api/v1/authn/factors/okta-poll/poll',
-              data: {
-                stateToken: '00_J1qxqyLs-6ZutUUWfbqm-1nqnW6n2o5z2wnBRHs'
-              }
-            });
-          });
+            done();
+          }, 9000);
```

Comment:
I am guessing we can reach 3 calls in 3 seconds itself. 

1) First call - takes us to poll screen 
2) Second call happens after timeout specificed ( 1sec) 
3) Third call happens after 1200ms. 
So in 2200ms 3 calls should have been made right.

Reply:
In theory yes, but I had to play around with it. I can update this to 5000 as thats the earliest it ended.